component/common/relabel/relabel.go

```diff
@@ -166,7 +170,7 @@ func (rc *Config) UnmarshalRiver(f func(interface{}) error) error {
 	}
 
 	if rc.Action == KeepEqual || rc.Action == DropEqual {
-		if rc.Regex != DefaultRelabelConfig.Regex ||
+		if !reflect.DeepEqual(*rc.Regex.Regexp, *DefaultRelabelConfig.Regex.Regexp) ||
```

Comment:
This was updated because the comparison was checking pointers previously rather than the values which won't work anymore.